modules/gltf/extensions/physics/gltf_physics_body.cpp

```diff
@@ -184,96 +211,138 @@ CollisionObject3D *GLTFPhysicsBody::to_node() const {
 Ref<GLTFPhysicsBody> GLTFPhysicsBody::from_dictionary(const Dictionary p_dictionary) {
 	Ref<GLTFPhysicsBody> physics_body;
 	physics_body.instantiate();
-	ERR_FAIL_COND_V_MSG(!p_dictionary.has("type"), physics_body, "Failed to parse GLTF physics body, missing required field 'type'.");
-	const String &body_type = p_dictionary["type"];
-	physics_body->body_type = body_type;
-
-	if (p_dictionary.has("mass")) {
-		physics_body->mass = p_dictionary["mass"];
+	Dictionary motion;
+	if (p_dictionary.has("motion")) {
+		motion = p_dictionary["motion"];
+	} else {
+		motion = p_dictionary;
+	}
+	if (motion.has("type")) {
+		// Read the body type. This representation sits between glTF's and Godot's physics nodes.
+		// While we may only read "static", "kinematic", or "dynamic" from a valid glTF file, we
+		// want to allow another extension to override this to another Godot node type mid-import.
+		// For example, a vehicle extension may want to override the body type to "vehicle"
+		// so Godot generates a VehicleBody3D node. Therefore we distinguish by importing
+		// "dynamic" as "rigid", and "kinematic" as "animatable", in the GLTFPhysicsBody code.
+		String body_type = motion["type"];
+		if (body_type == "dynamic") {
+			body_type = "rigid";
+		} else if (body_type == "kinematic") {
+			body_type = "animatable";
+		}
```

Comment:
This is confusing. If this is a string, it should mirror what is in the GLTF data model itself. If we want to map things to Godot-specific types, we should migrate it into an enum.

Reply:
It is a String-based enum.